src/lib/native.ts

```diff
@@ -343,8 +343,21 @@ export async function temp(content: string, prefix: string) {
     })
 }
 
-export async function move(from: string, to: string) {
-    return sendNativeMsg("move", { from, to }).catch(e => {
+export async function move(
+    from: string,
+    to: string,
+    overwrite: boolean,
+    cleanup: boolean,
+) {
+    const requireNativeMessengerVersion = "0.3.0"
+
+    if (!(await nativegate(requireNativeMessengerVersion, false))) {
```

Comment:
```suggestion
    if (overwrite && cleanup && !(await nativegate(requireNativeMessengerVersion, false))) {
```

`move` works fine without these new parameters

Reply:
@bovine3dom 

- `move` with `--overwrite` and/or `--cleanup` should work as expected in https://github.com/gsbabil/tridactyl_native_messenger/commit/043488b3cff5943678ed62bbbddda7ca3bf90ad6. 

-  https://github.com/gsbabil/tridactyl_native_messenger/commit/043488b3cff5943678ed62bbbddda7ca3bf90ad6 also fixes the notification problem by using `messageActiveTab("commandline_frame", "fillcmdline" ...` So we have success/failure notifications for the user if the download/move succeeded or failed. Thanks.

Reply:
Are you saying we should stop users who can currently use `:saveas` from using `:saveas` because `:saveas` will continue to not return success messages unless they update?

I don't think that's a good enough reason to force people to update. Some people with bizarre corporate security ~~theatre~~ policies may not be able to use the Nim binary and will have to continue to use the Python script, so I want to keep that supported as far as possible.

Am I missing something?

Reply:
@bovine3dom 

> Are you saying we should stop users who can currently use :saveas from using :saveas because :saveas will continue to not return success messages unless they update?

No. As mentioned in Matrix, my test-case was download attempts made on single files, and a notificaiton is useful from the user's perspective to know when/if the downloade completed successfully.

If this conflicts with multiple file download, I'm happy to migrate to a different notification approach where the user still gets to learn which file(s) downloaded property and which ones failed.

Reply:
@bovine3dom 

> Am I missing something?

As long as we are talking about a single file download, I don't fully understand where the conflict actually is. 

As said, my only goal from user's point of view is to notify when/if the download completed. I'm saying let's show the user something on "success", as well as on "error. The code before was only throwing error, no notification on "success".

Reply:
I think we're talking past each other. Without this patch, say I have native 0.1.11:

`:saveas foo` # works

With this patch, native 0.1.11

`:saveas foo` # doesn't work: you need to update your native messenger.

This is bad, right?

Reply:
@bovine3dom 

> I think we're talking past each other.

I thought you'd be pulling from my `master` where the latest currently points to https://github.com/gsbabil/tridactyl/commit/a508f432f20cdb875509402a8ee70d5aa3cdbaea

Can you you please confirm which commit you are reviewing? Thanks.

Reply:
@bovine3dom 

> With this patch, native 0.1.11

> `:saveas foo` # doesn't work: you need to update your native messenger.`

This should work. 

Let me check `0.1.11` native messenger with the `--overwrite` and `--cleanup` patches at https://github.com/gsbabil/tridactyl/commit/a508f432f20cdb875509402a8ee70d5aa3cdbaea. Thanks.

Reply:
https://github.com/gsbabil/tridactyl/commit/ee7e339951120e43fe5f807a704a793b8da316c7 is what I'm looking at

Reply:
@bovine3dom 

> gsbabil@ee7e339 is what I'm looking at

Please review https://github.com/gsbabil/tridactyl/commit/c17ac418266995de2f2540ee1178ee79eb41074a it should support both old `saveas` behaviour with Python and new `saveas` with Nim. Thanks.